packages/backend/src/util/validateRequest.ts

```diff
@@ -0,0 +1,35 @@
+import express from "express";
+import { APIValidator } from "@withorbit/api";
+
+type CurriedMiddleware = (validator: APIValidator) => express.RequestHandler;
+
+export const validateRequest: CurriedMiddleware = (validator) => (
+  req,
+  res,
+  next,
+) => {
+  convertNumbersToPrimitiveType(req.query);
+
+  const validationResult = validator.validateRequest({
+    method: req.method,
+    path: req.path,
+    query: req.query,
+    body: req.body,
+  });
+
+  if (validationResult === true) {
+    next();
+  } else {
+    res.status(400).send(validationResult);
+  }
+};
+
+function convertNumbersToPrimitiveType(object: Record<string, unknown>) {
```

Comment:
Daring! And slightly magic: could you add a bit of documentation describing what's going on here? I worry that in the future we may add some string query parameter to some API for which a client may _sometimes_ specify a number… and then the type for the same field on the same route may vary at runtime! (I don't suppose we can straightforwardly read the "intended" type via ajv at runtime?)

Reply:
I wasn't too keen on this code either. Your comment got me thinking, why doesn't `ajv` just do this inference? They already know which type it is supposed to be! Turns out, [they do](https://ajv.js.org/coercion.html), oops. I enabled the flag and got rid of this code.

Reply:
Wow, fantastic!!